transport/src/main/java/io/netty/channel/embedded/EmbeddedChannel.java

```diff
@@ -275,6 +281,13 @@ public boolean isActive() {
         return inboundMessages;
     }
 
+    private Queue<Pair<Object, ChannelPromise>> tempInboundMessages() {
+        if (tempInboundMessages == null) {
+            tempInboundMessages = new ArrayDeque<Pair<Object, ChannelPromise>>();
```

Comment:
I think we also need to ensure we release all `tmpInboundMessage` when `finish*` / `close` is called to ensure we not leak any memory.

Reply:
Fixed.
Since both `finish` and `close` will trigger `doClose`, drain `tempinboundmessages` in `doClose` method.